### DISCUSSION TITLE
Feature Request: Appender "append default" function

### DISCUSSION THREAD

**Discussion by JackMyers001** [duckdb/duckdb#6080] [Ideas]
*Note that all example code uses DuckDB.NET with the new .NET6 console application template, however I believe this code would be fairly similar with the official Java driver etc.*

As far as I'm aware, there's currently no `AUTO_INCREMENT` or `SERIAL` data type in DuckDB to create an auto-incrementing primary key. However, there is a workaround using the `SEQUENCE` option (as I learnt from [this StackOverflow question](https://stackoverflow.com/q/72883083)):

```csharp
using DuckDB.NET.Data;

var TABLE_NAME = "TestTable";
var NAMES = new List<string> { "Jim", "Fred", "Bob" };

using var db = new DuckDBConnection($"DataSource=:memory:");
db.Open();

var dbCommand = db.CreateCommand();

dbCommand.CommandText = "CREATE SEQUENCE seq_rowid START 1;";
dbCommand.ExecuteNonQuery();

dbCommand.CommandText = $"CREATE TABLE {TABLE_NAME}(RowId INTEGER PRIMARY KEY NOT NULL default nextval('seq_rowid'), Name VARCHAR);";
dbCommand.ExecuteNonQuery();
```

I can now add new entries into the table using `INSERT` statements:

```csharp
// Write values into database
foreach (var name in NAMES) {
    dbCommand.CommandText = $"INSERT INTO {TABLE_NAME} VALUES (DEFAULT, '{name}');";
    dbCommand.ExecuteNonQuery();
}

// Print out results
dbCommand.CommandText = $"SELECT * FROM {TABLE_NAME}";
using var reader = dbCommand.ExecuteReader();

var results = new List<string>();
while (reader.Read()) {
    var rowId = reader.GetInt64(0);
    var name = reader.GetString(1);
    Console.WriteLine($"{rowId}: {name}");
}

db.Close();
```

As expected, the output is

```text
1: Jim
2: Fred
3: Bob
```

However, according to the [DuckDB documentation on Insert Statements](https://duckdb.org/docs/data/insert), using insert statements for more than a few rows has "significant overhead". Even when using transactions as the docs suggest, I have found terrible performance (understandably). It is also possible to insert multiple rows in one SQL statement, however I don't believe that will increase performance by any significant margin.

Instead, I would like to use appenders. Unfortunately, I haven't found a way to add default values using appenders.

This is a hypothetical block of code I would like to write instead of using `INSERT` statements:

```csharp
using var appender = db.CreateAppender(TABLE_NAME);

// Write values into database
foreach (var name in NAMES) {
    var row = appender.CreateRow();
    row.AppendDefault().AppendValue(name);
    row.EndRow();
}

appender.Close();

// Print out results... (same as before)
```

Regardless of whether or not a value is a primary key / marked `NOT NULL`, I have not found a way to use appenders to insert rows (e.g. `row.AppendNull()`, `row.AppendValue("DEFAULT")` either throw DuckDBExceptions or have unintended behaviour (e.g. every `RowId` becomes 0 / null)).

**Comment by Tishj** [comment]
Definitely sounds like it's worth adding!
For now though, bundling your inserts into one statement should be many times faster than single row inserts
